app/controllers/api/orders_controller.rb

```diff
@@ -1,5 +1,10 @@
 module Api
   class OrdersController < BaseController
+    def show
+      authorize! :read, order
+      render json: @order, serializer: Api::OrderDetailedSerializer
```

Comment:
I would just call it `Api::OrderSerializer` so it maps one to one with the model and we avoid some boilerplate IMO. Or do we have a class call like that already?

Reply:
since you memoized the order below, you can use `order` instead of `@order` here.

Reply:
Yeah, we have Api::OrderSerializer, Api::Admin::OrderSerializer and now Api::OrderDetailedSerializer.
I think this is ok for such a central class (not if it was a less important one). The first one is for the shopfront, the second one for the admin but without much detail and this one includes everything as required in this PR.

We could potentially merge Api::OrderSerializer and Api::Admin::OrderSerializer in one. But I think that's out of the scope of this PR.

I changed from @order to order :+1:

Reply:
Also for another PR but perhaps we should rename Api::OrderSerializer to Api::ShopfrontOrderSerializer or something else that reflects the shopfront use case.